hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAbfsStreamStatistics.java

```diff
@@ -46,66 +46,89 @@ public void testAbfsStreamOps() throws Exception {
+      inForOneOperation = fs.open(smallOperationsFile);
+      inForOneOperation.read(testReadWriteOps.getBytes(), 0,
+          testReadWriteOps.getBytes().length);
+
+      //Test for one read and write operation
+      assertReadWriteOps(1, statistics);
+    } finally {
+      if (inForOneOperation != null) {
+        inForOneOperation.close();
+      }
+      if (outForOneOperation != null) {
+        outForOneOperation.close();
+      }
+    }
+    //Validating if content is being written in the smallOperationsFile
+    Assert.assertTrue("Mismatch in content validation",
+        validateContent(fs, smallOperationsFile,
             testReadWriteOps.getBytes()));
 
-    FSDataOutputStream outForLargeOperations = fs.create(largeOperationsFile);
-    statistics.reset();
-
+    FSDataOutputStream outForLargeOperations = null;
+    FSDataInputStream inForLargeOperations = null;
     StringBuilder largeOperationsValidationString = new StringBuilder();
-    for (int i = 0; i < 1000000; i++) {
-      outForLargeOperations.write(testReadWriteOps.getBytes());
-
-      //Creating the String for content Validation
-      largeOperationsValidationString.append(testReadWriteOps);
+    try {
+      outForLargeOperations = fs.create(largeOperationsFile);
+      statistics.reset();
+      int largeValue = 1000000;
+      for (int i = 0; i < largeValue; i++) {
+        outForLargeOperations.write(testReadWriteOps.getBytes());
+
+        //Creating the String for content Validation
+        largeOperationsValidationString.append(testReadWriteOps);
+      }
+
+      inForLargeOperations = fs.open(largeOperationsFile);
+      for (int i = 0; i < largeValue; i++)
+        inForLargeOperations
+            .read(testReadWriteOps.getBytes(), 0,
+                testReadWriteOps.getBytes().length);
+
+      //Test for one million read and write operations
+      assertReadWriteOps(largeValue, statistics);
+    } finally {
+      if (inForLargeOperations != null) {
+        inForLargeOperations.close();
+      }
+      if (outForLargeOperations != null) {
+        outForLargeOperations.close();
+      }
     }
 
-    FSDataInputStream inForLargeCalls = fs.open(largeOperationsFile);
+    //Validating if content is being written in largeOperationsFile
+    Assert.assertTrue("Mismatch in content validation",
+        validateContent(fs, largeOperationsFile,
+            largeOperationsValidationString.toString().getBytes()));
 
-    for (int i = 0; i < 1000000; i++)
-      inForLargeCalls
-          .read(testReadWriteOps.getBytes(), 0,
-              testReadWriteOps.getBytes().length);
+  }
 
-    //Test for one million read and write operations
-    Assert.assertEquals("Mismatch in read operations", 1000000,
+  /**
+   * Method for Read and Write Ops Assertion.
+   *
+   * @param expectedReadWriteOps Expected Value
+   * @param statistics           fs stats to get Actual Values
+   */
+  private void assertReadWriteOps(long expectedReadWriteOps,
```

Comment:
This is wrong. How can you compare the expected value with both read and write value? What if read and write values are different?
 assertReadWriteOps(operation, expected, actual) is what we discussed right?